pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -270,6 +270,64 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 			}
 		}
 	}
+
+	// Deploying test based on the storageclassDeviceSet
```

Comment:
nit: could you split the device set handling into a separate method? it wouldn't hurt to create a separate method for the above `for` loop as well to keep the methods smaller. this will hopefully make it easier to unit test as well

Reply:
yup correct